padertorch/configurable.py

```diff
@@ -325,15 +369,18 @@ def get_config(
         else:
             config = _sacred_dogmatic_to_dict(updates)
 
-        if 'factory' not in config:
-            config['factory'] = cls
-        else:
+        if 'factory' in config:
             config['factory'] = import_class(config['factory'])
             if inspect.isclass(config['factory']) \
                     and issubclass(config['factory'], Configurable):
                 # When subclass of Configurable expect proper subclass
-                assert issubclass(import_class(config['factory']), cls), (
+                assert issubclass(config['factory'], cls), (
                     config['factory'], cls)
+        elif 'partial' in config:
+            config['partial'] = import_class(config['partial'])
+            assert callable(config['partial']), config['partial']
```

Comment:
Not if we want to allow functions here

Reply:
Functions are allowed in `factory` as well, so that's not a problem. The check checks that if `factory` is `Configurable`, then it should be a sub-class of `cls`, so that `A.from_config(...)` always returns an instance of a sub-class of `A` if `factory` is a `Configurable`.